incubator/jaeger/templates/collector-svc.yaml

```diff
@@ -1,35 +1,49 @@
+{{- if .Values.collector.enabled -}}
 apiVersion: v1
 kind: Service
 metadata:
-  name: "{{ template "jaeger.fullname" . }}-collector"
+  name: {{ template "jaeger.fullname" . }}-collector
   labels:
-    app: "{{ template "jaeger.name" . }}"
+    app: {{ template "jaeger.name" . }}
     jaeger-infra: collector-service
     chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
-    component: "collector"
-    heritage: "{{ .Release.Service }}"
-    release: "{{ .Release.Name }}"
+    component: collector
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
 {{- if .Values.collector.service.annotations }}
   annotations:
-{{ toYaml .Values.collector.service.annotations | indent 6 }}
+{{ toYaml .Values.collector.service.annotations | indent 4 }}
 {{- end }}
 spec:
   ports:
   - name: jaeger-collector-tchannel
     port: {{ .Values.collector.service.tchannelPort }}
     protocol: TCP
+    {{- if eq .Values.collector.service.type "NodePort" }}
+    nodePort: {{ .Values.collector.service.tchannelPort }}
```

Comment:
Is there any reason why we are overriding the NodePort's port?   
By default it will be assigned to something between 30000-32767 (https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport)

This would mean that only a single Collector can run on the default port per Kubernetes cluster.

Reply:
The reason we need this is for cases when the agent is on one cluster and the collector is on another and we need a predefined port.

Reply:
Gotcha! On our side, we are using `type: LoadBalancer` for that. 👍

Reply:
This means that if you are sending tracing in cluster, you are going to create a load balancer. This is totally unnecessary for an in cluster solution and ends up costing money.

Reply:
No, a Service with type LoadBalancer also has a ClusterIP that can be reached from within the cluster directly (no outbound traffic at all) 👍